### PR TITLE
added 'Reference - Dependency Injection Container' documentation

### PR DESCRIPTION
**Pull request by eric-chau**
Referenced issue: [backbee/backbee-standard #645](https://github.com/backbee/backbee-standard/issues/645)

ping @mickaelsteinberg, @crouillon.

### CODE REVIEW

**Comment by crouillon** [comment]
Very clear, thank you @eric-chau  :+1:

File: doc/developer-documentation/references/dependency-injection-container.md [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+$container->get('classcontent.manager'); // returns an instance of BackBee\ClassContent\ClassContentManager
+
+$container->getParameter('debug'); // returns true or false, depending on your configuration
+```
+
+You can see all parameters and services registered by BackBee by exploring every YAML files inside the `Config/services/` folder (or `vendor/backbee/backbee/Config/services/` if you're using BackBee Standard Edition).
+
+## To register new service and parameter
+
+The easiest way for you to register a new service or a parameter inside BackBee's service container is to use configurations files. Even if you can use XML or YAML format, we highly recommend you to use YAML cause it's more simple and readable. To learn about YAML syntax with [its official website](http://yaml.org/).
```
Comment (mickaelsteinberg) [inline doc/developer-documentation/references/dependency-injection-container.md]: - "is to use ~~configurations~~ configuration files"
- "Even if [...] readable" => I would replace it with "Even if XML and YAML formats are accepted, we highly recommend to use YAML for its readability and simplicity."
- "~~To~~ Learn more about YAML syntax ~~with~~ **at** [...]"

File: doc/developer-documentation/references/dependency-injection-container.md [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+
+The setting file must be named `services.yml` and be in one of these locations:
```
Comment (mickaelsteinberg) [inline doc/developer-documentation/references/dependency-injection-container.md]: - ~~setting~~ **settings**

File: doc/developer-documentation/references/dependency-injection-container.md [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+services:
+    current_datetime:
+        class: %current_datetime.class%
+        arguments: ["now"]
+
+    foobar:
+        class: \Foobar
+        arguments: ["@current_datetime", "%name%"]
+```
+
```
Comment (mickaelsteinberg) [inline doc/developer-documentation/references/dependency-injection-container.md]: Did I miss something or you didn't show how to use `@__IDENTIFIER__` in this example?

**Comment by mickaelsteinberg** [comment]
Reviewed ! Great work though, really liked how you broke down the subject piece by piece and made DIC simple to understand

File: doc/developer-documentation/references/dependency-injection-container.md (hunk 3 above)
Comment (eric-chau) [inline doc/developer-documentation/references/dependency-injection-container.md]: The `Inject a service or a parameter inside a service constructor` section is supposed to explain it. Is it not enough clear? Let me know.

ping @mickaelsteinberg

**Comment by eric-chau** [comment]
thank you for your review @mickaelsteinberg!

File: doc/developer-documentation/references/dependency-injection-container.md (hunk 3 above)
Comment (mickaelsteinberg) [inline doc/developer-documentation/references/dependency-injection-container.md]: Got it! I saw you used it in the YAML file as an example. The trick is I thought the identifier was supposed to be written uppercase with a `@` and surrounded by 2 `_` 
That's what I missed I guess

**Comment by mickaelsteinberg** [comment]
@eric-chau : NP ! I also added a quick comment

**Pull request merged**